src/main/java/uk/gov/hmcts/reform/camunda/bpm/filter/SpringSecurityApiAuthenticationProvider.java

```diff
@@ -0,0 +1,96 @@
+import org.camunda.bpm.engine.identity.User;
+import org.camunda.bpm.engine.rest.security.auth.AuthenticationResult;
+import uk.gov.hmcts.reform.authorisation.exceptions.InvalidTokenException;
+import uk.gov.hmcts.reform.authorisation.exceptions.ServiceException;
+import uk.gov.hmcts.reform.authorisation.validators.AuthTokenValidator;
+import uk.gov.hmcts.reform.camunda.bpm.app.AuthorizationHelper;
+import uk.gov.hmcts.reform.camunda.bpm.config.ConfigProperties;
+import uk.gov.hmcts.reform.camunda.bpm.config.GroupConfig;
+import uk.gov.hmcts.reform.camunda.bpm.context.SpringContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+
+@SuppressWarnings("unused")
+public class SpringSecurityApiAuthenticationProvider extends SpringSecurityBaseAuthenticationProvider {
+
+    public static final String AUTHORISATION = "ServiceAuthorization";
+    
+    @Override
+    @SuppressWarnings("unchecked")
+    public AuthenticationResult extractAuthenticatedUser(HttpServletRequest request, ProcessEngine engine) {
+
+        try {
+            AuthTokenValidator authTokenValidator = SpringContext.getAppContext().getBean(AuthTokenValidator.class);
+            String bearerToken = extractBearerToken(request);
+            String serviceName = authTokenValidator.getServiceName(bearerToken);
+            configProperties = SpringContext.getAppContext().getBean(ConfigProperties.class);
+            AuthenticationResult authenticationResult = new AuthenticationResult(
+                serviceName,
+                true
+            );
+
+            IdentityService identityService = engine.getIdentityService();
+            updateUser(serviceName, identityService);
+
+            List<GroupConfig> applicableGroups = getCamundaGroupsList(serviceName);
+
+            authenticationResult.setTenants(getTenantsAndProvision(serviceName, applicableGroups, identityService));
+            List<String> camundaGroups = getCamundaGroupsAndProvision(serviceName, applicableGroups, identityService);
+            authenticationResult.setGroups(camundaGroups);
+
+            if (authenticationResult.getTenants().isEmpty()) {
+                return AuthenticationResult.unsuccessful();
+            }
+
+            AuthorizationService authorizationService = engine.getAuthorizationService();
+            authorizationService
+                .createNewAuthorization(Authorization.AUTH_TYPE_GRANT);
+
+            AuthorizationHelper authorizationHelper = new AuthorizationHelper(
+                engine.getAuthorizationService());
+
+            refreshAuthorisation(authorizationHelper);
+            return authenticationResult;
+        } catch (InvalidTokenException | ServiceException exception) {
+            return AuthenticationResult.unsuccessful();
+        }
+
+    }
+
+    private String extractBearerToken(HttpServletRequest request) {
+        String token = request.getHeader(AUTHORISATION);
```

Comment:
is it better to clarify that this is service authorisation?
whenever i see `request.getHeader(AUTHORISATION)` I assume that constant refers to the `Authorization` header?
```suggestion
        String token = request.getHeader(SERVICE_AUTHORISATION);
```